app/models/concerns/user_access_key_overrides.rb

```diff
@@ -30,20 +19,21 @@ def valid_password?(password)
   end
 
   def unlock_user_access_key(password)
-    self.user_access_key = Encryption::UserAccessKey.new(
-      password: password,
-      salt: authenticatable_salt,
-      cost: password_cost
-    ).unlock(encryption_key)
+    self.user_access_key = build_user_access_key(password).unlock(encryption_key)
   end
 
   def password=(new_password)
-    if new_password.present?
-      self.password_salt = Devise.friendly_token[0, 20]
-      self.encryption_key = nil
-      self.password_cost = nil
-    end
-    super
+    @password = new_password
+    return unless new_password.present?
+
+    self.password_salt = Devise.friendly_token[0, 20]
+
+    user_access_key = build_user_access_key(new_password, cost: nil).build
```

Comment:
should this be `self.user_access_key`? I left as it was before.

Reply:
We did this b/c we used the same key for PII encryption, but we don't do that anymore, so this shouldn't be an issue.

Reply:
I wonder if you wouldn't want to extract some of the above from this (`password`) setter into a separate method, to better encapsulate the intent...